torchgen/gen_executorch.py

```diff
@@ -168,8 +168,13 @@ def __call__(
         kernel_key: Union[ETKernelKey, List[ETKernelKey]] = unbox_kernel_entry[1][0]
         kernel_meta: BackendMetadata = unbox_kernel_entry[1][1]
 
-        # TODO: Update to use Kernel Selector
-        if not self.selector.is_root_operator(f"{f.namespace}::{f.func.name}"):
+        op_name = f.namespace + "::" + str(f.func.name)
+        if not isinstance(kernel_key, list):
+            kernel_key = [kernel_key]
+        used_kernel_keys = self.selector.et_get_selected_kernels(
+            op_name, [k.to_native_string() for k in kernel_key]
+        )
+        if not used_kernel_keys or not self.selector.is_root_operator(op_name):
```

Comment:
Check `is_root_operator()` earlier and skip quickly. Don't need to compute kernel key if not root operator.